src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,977 @@
+                            if (typeCheckMethod(_semanticModel.GetTypeInfo(((VariableDeclarationSyntax)variableDeclarator.Parent).Type).Type))
+                            {
+                                // before 
+                                // var a = (from a in b select a).ToList();
+                                // after 
+                                // var a = new List<T>();
+                                // foreach(...)
+                                variable = SyntaxFactory.IdentifierName(variableDeclarator.Identifier);
+                                nodesBefore = new[] { parentStatement.ReplaceNode(invocationExpression, initializer.WithAdditionalAnnotations(Simplifier.Annotation)) };
+                                nodesAfter = new CSharpSyntaxNode[] { };
+                            }
+                            else
+                            {
+                                // before 
+                                // IReadOnlyList<int> a = (from a in b select a).ToList();
+                                // after 
+                                // var list = new List<T>();
+                                // foreach(...)
+                                // IReadOnlyList<int> a = list;
+                                variable = SyntaxFactory.IdentifierName(symbolName);
+                                nodesBefore = new[] { CreateLocalDeclarationStatement(symbolName, initializer) };
+                                nodesAfter = new CSharpSyntaxNode[] { parentStatement.ReplaceNode(invocationExpression, variable.WithAdditionalAnnotations(Simplifier.Annotation)) };
+                            }
+
+                            return true;
+                        }
+
+                        break;
+                    case SyntaxKind.SimpleAssignmentExpression:
+                        var assignmentExpression = (AssignmentExpressionSyntax)invocationParent;
+                        if (assignmentExpression.Right.WalkDownParentheses() == invocationExpression)
+                        {
+                            // Check that we can re-use the local variable or parameter
+                            if (IsLocalOrParameterSymbol(assignmentExpression.Left) &&
+                                typeCheckMethod(_semanticModel.GetTypeInfo(assignmentExpression.Left, _cancellationToken).Type))
+                            {
+                                // before 
+                                // a = (from a in b select a).ToList();
+                                // after 
+                                // a = new List<T>();
+                                // foreach(...)
+                                variable = assignmentExpression.Left;
+                                nodesBefore = new[] { parentStatement.ReplaceNode(invocationExpression, initializer.WithAdditionalAnnotations(Simplifier.Annotation)) };
+                                nodesAfter = new CSharpSyntaxNode[] { };
```

Comment:
>nodesAfter = new CSharpSyntaxNode[] { }; [](start = 32, length = 40)

It looks like this transformation is very similar to the one from the previous case, consider extracting into a helper method and reusing it.